### PR TITLE
[MIG] hr_employee_firstname: Migration to 17.0 t#76430

### PR DESCRIPTION
**Pull request by andreagidaltig**
[Task #76430](https://www.vauxoo.com/web#id=76430&cids=1&menu_id=1530&model=project.task&view_type=form)
- Update definition of required modifier as part of [1].
- Update incoming parameters of post_init_hook method since `cr, registry`
  were replaced by `env` in [2].
- Deprecate the use of _onchange_spec method and use the method new method
  _get_fields_spec to return the fields specification from a view
  description since it was introduced in [3].
  
[1] https://github.com/odoo/odoo/pull/104741
[2] https://github.com/odoo/odoo/commit/b4a7996
[3] https://github.com/odoo/odoo/commit/f5e6494

### CODE REVIEW

**Comment by andreagidaltig** [comment]
Hi @luistorresm could you review this please?

**Comment by luistorresm** [comment]
Hello @andreagidaltig 

Could you join the commits related, please?

More info [here](https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate)

cc @luisg123v

**Comment by andreagidaltig** [comment]
Hi @luistorresm 

I verified this with @luisg123v  and I just merged some translation commits from same author and same language as is indicated [here](https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate).

**Comment by luistorresm** [comment]
👍 

@luisg123v 

Could you merge, please?

Regards

File: hr_employee_firstname/tests/test_hr_employee_firstname.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        self.assertEqual(self.employee2_id.name, "Jean-Pierre Carnaud")
+
+        # Check for employee3
+        self.assertEqual(self.employee3_id.name, "Famke Jenssens")
+
+    def test_onchange(self):
+        """
+        Validate the get_name method is not failing
+        """
+        field_onchange = self.env["hr.employee"].new({})._get_fields_spec()
```
Comment (luisg123v) [inline hr_employee_firstname/tests/test_hr_employee_firstname.py]: This is not an onchange anymore AFAIK, so I think you shouldn't neither name it as so nor using `new()`.

File: hr_employee_firstname/tests/test_hr_employee_firstname.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+        self.assertEqual(
+            submap(field_onchange, ('firstname', 'lastname')),
```
Comment (luisg123v) [inline hr_employee_firstname/tests/test_hr_employee_firstname.py]: Why this?

**Review by luisg123v — CHANGES_REQUESTED** [review]

**Comment by luisg123v** [comment]
Missing task ID on PR's title.

**Comment by andreagidaltig** [comment]
Hi @luisg123v I just modified this, could you review it again please?

cc. @luistorresm

**Review by luisg123v — APPROVED** [review]
LGTM :+1:

**Comment by luisg123v** [comment]
Be aware once the OCA releases their branches, this will get overwritten and we will need to make a new PR targeting the OCA.

**Pull request merged**